### PR TITLE
Two tiny doc things, combine regression fix

### LINKED ISSUE #14312: `[combine` texture modifier regression
```
Minetest 5.9.0-dev-89f3502b56 (Linux)
Using Irrlicht 1.9.0mt14
Using LuaJIT 2.1.1692716794
Running on Linux/6.6.6-200.fc39.x86_64 x86_64
BUILD_TYPE=RelWithDebInfo
RUN_IN_PLACE=1
USE_CURL=1
USE_GETTEXT=1
USE_SOUND=1
STATIC_SHAREDIR="."
STATIC_LOCALEDIR="locale"
```

**Steps to reproduce**

1. Start a MineClone 2 game
2. Execute `/giveme mcl_heads:creeper`
3. Place it

**Result**

Broken rendering
screenshot broken

```
generateImagePart(): invalid X or Y offset for part_of_name="[combine:16x16:-36,4=mcl_heads_creeper.png", cancelling.
generateImage(): Failed to generate "[combine:16x16:-36,4=mcl_heads_creeper.png"
generateImagePart(): invalid X or Y offset for part_of_name="[combine:16x16:-36,4=mcl_heads_creeper.png", cancelling.
generateImage(): Failed to generate "[combine:16x16:-36,4=mcl_heads_creeper.png"
generateImagePart(): invalid X or Y offset for part_of_name="[combine:16x16:-44,4=mcl_heads_creeper.png", cancelling.
generateImage(): Failed to generate "[combine:16x16:-44,4=mcl_heads_creeper.png"
generateImagePart(): invalid X or Y offset for part_of_name="[combine:16x16:-28,0=mcl_heads_creeper.png", cancelling.
generateImage(): Failed to generate "[combine:16x16:-28,0=mcl_heads_creeper.png"
generateImagePart(): invalid X or Y offset for part_of_name="[combine:16x16:-44,0=mcl_heads_creeper.png", cancelling.
generateImage(): Failed to generate "[combine:16x16:-44,0=mcl_heads_creeper.png"
generateImagePart(): invalid X or Y offset for part_of_name="[combine:16x16:-52,0=mcl_heads_creeper.png", cancelling.
generateImage(): Failed to generate "[combine:16x16:-52,0=mcl_heads_creeper.png"
generateImagePart(): invalid X or Y offset for part_of_name="[combine:16x16:-36,0=mcl_heads_creeper.png", cancelling.
generateImage(): Failed to generate "[combine:16x16:-36,0=mcl_heads_creeper.png"
generateImagePart(): invalid X or Y offset for part_of_name="[combine:16x16:-36,-4=mcl_heads_creeper.png", cancelling.
generateImage(): Failed to generate "[combine:16x16:-36,-4=mcl_heads_creeper.png"
generateImagePart(): invalid X or Y offset for part_of_name="[combine:16x16:-52,-4=mcl_heads_creeper.png", cancelling.
generateImage(): Failed to generate "[combine:16x16:-52,-4=mcl_heads_creeper.png"
generateImagePart(): invalid X or Y offset for part_of_name="[combine:16x16:-40,-4=mcl_heads_creeper.png", cancelling.
generateImage(): Failed to generate "[combine:16x16:-40,-4=mcl_heads_creeper.png"
generateImagePart(): invalid X or Y offset for part_of_name="[combine:16x16:-32,-4=mcl_heads_creeper.png", cancelling.
generateImage(): Failed to generate "[combine:16x16:-32,-4=mcl_heads_creeper.png"
generateImagePart(): invalid X or Y offset for part_of_name="[combine:16x16:-36,0=mcl_heads_creeper.png", cancelling.
generateImage(): Failed to generate "[combine:16x16:-36,0=mcl_heads_creeper.png"
generateImagePart(): invalid X or Y offset for part_of_name="[combine:16x16:-36,0=mcl_heads_creeper.png", cancelling.
generateImage(): Failed to generate "[combine:16x16:-36,0=mcl_heads_creeper.png"
generateImagePart(): invalid X or Y offset for part_of_name="[combine:16x16:-44,8=mcl_heads_creeper.png", cancelling.
generateImage(): Failed to generate "[combine:16x16:-44,8=mcl_heads_creeper.png"
...
```

Working rendering (with b7822e705de7461c548ed5ed715121646bd6f626 reverted)
screenshot working

Looks like it doesn't accept negative offsets anymore?

### PR DESCRIPTION
**Pull request by appgurueu**
Fixes #14312 (last commit), among two doc improvements.

## How to test

Observe that the regression has regressed. Read the docs and commit messages.

### CODE REVIEW

File: doc/lua_api.md [hunk 1]
```diff
@@ -8402,6 +8402,7 @@ Player properties need to be saved manually.
     pointable = true,
     -- Can be `true` if it is pointable, `false` if it can be pointed through,
     -- or `"blocking"` if it is pointable but not selectable.
+    -- Older clients interpret `pointable = "blocking"` as `pointable = true`.
```
Comment (grorp) [inline doc/lua_api.md:8405]:
```suggestion
    -- Clients older than 5.9.0 interpret `pointable = "blocking"` as `pointable = true`.
```
also applies below

**Review by grorp — COMMENTED** [review]
The fix for the regression works.

The two documentation improvements could be squashed into one "Minor documentation fixes" commit.

File: src/client/tile.cpp [hunk 2]
```diff
@@ -1330,12 +1330,10 @@ bool TextureSource::generateImagePart(std::string part_of_name,
 				baseimg->fill(video::SColor(0,0,0,0));
 			}
 			while (!sf.at_end()) {
```
Comment (sfan5) [inline src/client/tile.cpp:1332]: there's another edge case above here that probably nobody actually uses:
combine inherits the parent dimensions in which case it's totally "valid" to set them to zero or 99999
code should be
```cpp
			u32 h0 = stoi(sf.next(":"));
			if (!baseimg) {
				CHECK_DIM(w0, h0);
				baseimg = driver->createImage(video::ECF_A8R8G8B8, {w0, h0});
				baseimg->fill(video::SColor(0,0,0,0));
			}
```

File: doc/lua_api.md [hunk 3]
```diff
@@ -613,13 +613,13 @@ Creates an inventorycube with `grass.png`, `dirt.png^grass_side.png` and
 * `<y>`: y position
 * `<color>`: a `ColorString`.
```
Comment (sfan5) [inline doc/lua_api.md:615]: if you're doing this anyway please document that [combine x or y can be negative

**Comment by sfan5** [comment]
Regression fix replaced as discussed on [IRC](https://irc.minetest.net/minetest-dev/2024-01-27).

**Comment by appgurueu** [comment]
I haven't tested the last commit well yet, but the logic should check out, and stuff (like signs) still looks right.

**Review by sfan5 — APPROVED** [review]
I will trust your testing

**Comment by appgurueu** [comment]
Tested. Code:

```lua
minetest.after(1, function()
	local E = minetest.formspec_escape
	local t = {"size[4,4]"}
	for x = 0, 1 do
		for y = 0, 1 do
			local tmod = ("[combine:2x2:0,0=rgbv.png:%d,%d=rgbv.png"):format(2*x-1, 2*y-1)
			table.insert(t, ("image[%d,%d;1,1;%s]"):format(x, y, E(tmod)))
		end
	end
	minetest.show_formspec("singleplayer", "dev:test", table.concat(t))
end)
```

Screenshot

[`rgbv.png`](https://github.com/minetest/minetest/assets/34514239/26c981ba-e59e-4b6b-8b85-f8917ee0b2d6)

**Pull request merged**